humble-ci-testing + clang-tidy: Gmock deprecation

### Description

CI is failing due to a deprecation in gmock [here](https://github.com/ros-planning/moveit2/actions/runs/5715687575/job/15485573319?pr=2277#step:12:5040)

```
In file included from /opt/ros/humble/src/gmock_vendor/src/gmock_main.cc:32:
In file included from /opt/ros/humble/src/gmock_vendor/include/gmock/gmock.h:61:
In file included from /opt/ros/humble/src/gmock_vendor/include/gmock/gmock-function-mocker.h:39:
In file included from /opt/ros/humble/src/gmock_vendor/include/gmock/gmock-generated-function-mockers.h:47:
In file included from /opt/ros/humble/src/gmock_vendor/include/gmock/gmock-spec-builders.h:75:
/opt/ros/humble/src/gmock_vendor/include/gmock/gmock-matchers.h:721:3: error: definition of implicit copy constructor for 'StrEqualityMatcher<std::basic_string<char>>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  GTEST_DISALLOW_ASSIGN_(StrEqualityMatcher);
  ^ 
/opt/ros/humble/src/gtest_vendor/include/gtest/internal/gtest-port.h:682:8: note: expanded from macro 'GTEST_DISALLOW_ASSIGN_'
  void operator=(type const &) = delete
       ^ 
/opt/ros/humble/src/gtest_vendor/include/gtest/gtest-matchers.h:456:54: note: in implicit copy constructor for 'testing::internal::StrEqualityMatcher<std::basic_string<char>>' first required here
  explicit PolymorphicMatcher(const Impl& an_impl) : impl_(an_impl) {}
                                                     ^
/opt/ros/humble/src/gtest_vendor/include/gtest/gtest-matchers.h:514:10: note: in instantiation of member function 'testing::PolymorphicMatcher<testing::internal::StrEqualityMatcher<std::basic_string<char>>>::PolymorphicMatcher' requested here
  return PolymorphicMatcher<Impl>(impl);
         ^ 
/opt/ros/humble/src/gmock_vendor/include/gmock/gmock-matchers.h:3813:10: note: in instantiation of function template specialization 'testing::MakePolymorphicMatcher<testing::internal::StrEqualityMatcher<std::basic_string<char>>>' requested here
  return MakePolymorphicMatcher(

```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -23,9 +23,9 @@ jobs:
             ROS_DISTRO: humble
             IKFAST_TEST: true
             CLANG_TIDY: pedantic
-            # Silent gmock/gtest warnings: https://github.com/ament/googletest/pull/21
+            # Silent gmock/gtest warnings by picking more recent googletest version
             AFTER_BUILD_UPSTREAM_WORKSPACE: |
-              git clone --quiet --branch clalancette/update-to-gtest-1.11 https://github.com/ament/googletest "${BASEDIR}/upstream_ws/src/googletest"
+              git clone --depth 1 --quiet --branch 1.11.9000 https://github.com/ament/googletest "${BASEDIR}/upstream_ws/src/googletest"
               builder_run_build "/opt/ros/${ROS_DISTRO}" "${BASEDIR}/upstream_ws" --packages-select gtest_vendor gmock_vendor
     env:
       CXXFLAGS: >-

